src/components/Dropdown.tsx

```diff
@@ -0,0 +1,288 @@
+  | {
+    type: ActionType.ToggleDropdown,
+  }
+);
+
+/**
+ * Reducer that executes actions
+ * @author Alessandra De Lucas
+ * @param state current state
+ * @param action action to execute
+ */
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case ActionType.ToggleDropdown: {
+      return {
+        ...state,
+        isDropdownOpen: !state.isDropdownOpen,
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+/*------------------------------------------------------------------------*/
+/* ------------------------------ Component ----------------------------- */
+/*------------------------------------------------------------------------*/
+
+const Dropdown: React.FC<Props> = (props) => {
+  /*------------------------------------------------------------------------*/
+  /* -------------------------------- Setup ------------------------------- */
+  /*------------------------------------------------------------------------*/
+
+  /* -------------- Props ------------- */
+
+  // Destructure all props
+  const {
+    dropdownButton,
+    items,
+    variant,
+  } = props;
+
+  /* -------------- State ------------- */
+
+  // Initial state
+  const initialState: State = {
+    isDropdownOpen: false,
+  };
+
+  // Initialize state
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  // Destructure common state
+  const {
+    isDropdownOpen,
+  } = state;
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+      dispatch({ type: ActionType.ToggleDropdown });
+    }
+  };
+
+  /*------------------------------------------------------------------------*/
+  /* ------------------------- Lifecycle Functions ------------------------ */
+  /*------------------------------------------------------------------------*/
+
+  /**
+   * Mount
+   * @author Yuen Ler Chow
+   */
+  useEffect(() => {
+    // Add event listener to close dropdown when clicking outside
+    if (isDropdownOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      // Remove event listener when dropdown is closed
+      document.removeEventListener('mousedown', handleClickOutside);
+    }
+
+    // Cleanup
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isDropdownOpen]);
+
+  /*----------------------------------------*/
+  /* --------------- Main UI -------------- */
+  /*----------------------------------------*/
+
+  return (
+    <div className="dropdown" ref={dropdownRef}>
+      <button
+        className={
+          combineClassNames([
+            'btn dropdown-toggle border',
+            isDropdownOpen && 'show',
+            variant && variant === Variant.Dark && 'btn-dark text-light',
+            variant && variant === Variant.Light && 'btn-light text-dark',
```

Comment:
First, simplify this by having a default variant so `variant` is always defined (default to `Variant.Secondary`).

Second, build this dynamically (you only need to add a text color if the button is light).

Something like this:

```tsx
combineClassNames([
            'btn dropdown-toggle border',
            isDropdownOpen && 'show',
            `btn-${variant}`,
            variant === Variant.Light && 'text-dark',
```